src/main/kotlin/org/elm/lang/core/types/TypeInference.kt

```diff
@@ -117,8 +117,7 @@ private class InferenceScope(
         // If the assignee has any syntax errors, we don't run inference on it. Trying to resolve
         // references in bodies when the parameters contain syntax errors can result in infinite
         // recursion and surprising bugs.
-        // We don't currently infer recursive functions in order to avoid infinite loops in the inference.
-        if (checkRecursion(declaration) || assignee == null || assignee.hasErrors) {
+        if (assignee == null || assignee.hasErrors) {
```

Comment:
This check was superfluous with the check  in `inferReferencedValueDeclaration`.